@xen-orchestra/fs/src/s3.js

```diff
@@ -224,17 +224,34 @@ export default class S3Handler extends RemoteHandlerAbstract {
   }
 
   async _outputStream(path, input, { validator }) {
+    const MAX_PART = 10000
+    const PART_SIZE = 5 * 1024 * 1024
+    let readCounter = 0
+    const streamCutter = new Transform({
+      transform(chunk, encoding, callback) {
+        const MAX_SIZE = MAX_PART * PART_SIZE
+        readCounter += chunk.length
+        if (readCounter > MAX_SIZE) {
+          callback(new Error(`read ${readCounter} bytes, maximum size allowed  is ${MAX_SIZE} `))
+        } else {
+          callback(null, chunk)
+        }
+      },
+    })
+
     // Workaround for "ReferenceError: ReadableStream is not defined"
     // https://github.com/aws/aws-sdk-js-v3/issues/2522
     const Body = new PassThrough()
-    pipeline(input, Body, () => {})
+    pipeline(input, streamCutter, Body, () => {})
```

Comment:
@julien-f I wonder if here we do not "eat" an eventual error . I will test it on monday, it may lead to remove the streamcutter

Reply:
Tested and `pipeline` correctly forwards the error.